### PR TITLE
feat: address aliasing

### LINKED ISSUE #1229: design: anti-L1-spoofing messaging system
## Feature Request

The current Messaging System takes the l1_sender address as the caller of the L2 contract invoked.

This can be an issue if, for example, a L2 contract expects to be invoked in specific conditions by another L2 contract, but ends up being called in other conditions by an L1 contract that would match these conditions.

We need to design a system that prevents this from happening:
- Arbitrum has an offset on all L1 addresses executing messages on L2
- Optimism routes all messages from the XDomainMessenger Contract

### PR DESCRIPTION
**Pull request by obatirou**
Time spent on this PR:

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #1229

## What is the new behavior?

- adds address aliasing for L1 to L2 message

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot/1430)

### CODE REVIEW

File: solidity_contracts/src/L1L2Messaging/L1KakarotMessaging.sol [hunk 1]
```diff
@@ -28,7 +32,7 @@ contract L1KakarotMessaging {
     function sendMessageToL2(address to, uint248 value, bytes calldata data) external payable {
         uint256 totalLength = data.length + 4;
         uint256[] memory convertedData = new uint256[](totalLength);
-        convertedData[0] = uint256(uint160(msg.sender));
+        convertedData[0] = uint256(uint160(AddressAliasHelper.applyL1ToL2Alias(msg.sender)));
```
Comment (obatirou) [inline solidity_contracts/src/L1L2Messaging/L1KakarotMessaging.sol:35]: other solution suggested by @enitrat would be to do the aliasing inside Kakarot

**Review by enitrat — APPROVED** [review]
lgtm

File: solidity_contracts/src/L1L2Messaging/L1KakarotMessaging.sol (hunk 1 above)
Comment (ClementWalter) [reply solidity_contracts/src/L1L2Messaging/L1KakarotMessaging.sol:35]: let's keep this mechanism in pure solidity

**Review by ClementWalter — APPROVED** [review]

**Comment by ClementWalter** [comment]
doc issue is https://github.com/kkrt-labs/kakarot-docs/issues/77

**Pull request merged**